documentation/docs/fundamentals/performant-components.md

```diff
@@ -7,7 +7,26 @@ While `FlashList` does its best to achieve high performance, it will still perfo
 
 ## Recycling
 
-One important thing to understand is how `FlashList` works under the hood. When an item gets out of the viewport, instead of being destroyed, the component is re-rendered with a different `item` prop. When optimizing your item component, try to ensure as few things as possible have to be re-rendered and recomputed when recycling.
+One important thing to understand is how `FlashList` works under the hood. When an item gets out of the viewport, instead of being destroyed, the component is re-rendered with a different `item` prop. For example, if you make use of `useState` in a reused component, you may see state values that were set for that component when it was associated with a different item in the list, and would then need to reset any previously set state when a new item is rendered:
+
+```tsx
+const MyItem = ({ item }) => {
+  const lastItemId = useRef(item.someId);
+  const [liked, setLiked] = useState(item.liked);
+  if (item.someId !== lastItemId.current) {
```

Comment:
Let's also link the react docs mentioning this as being the alternative `getDerivedStateFromProps`. I also think we should mention that useState should be used ONLY if necessary and not having states is preferable.